### DISCUSSION TITLE
The difference between 'confidence_threshold' and 'match_conf'?

### DISCUSSION THREAD

**Discussion by Sherlock-IT** [OpenStitching/stitching#103] [Q&A] [answered]
what is the difference between 'confidence_threshold' and 'match_conf'.

**Comment by 3nol** [comment, marked as answer]
The naming can be a bit confusing, and indeed, both describe some form of confidence. You can imagine confidence as a filter for when to keep or when to throw away something. This _something_ is where these parameters are different:
- `match_conf` is passed to the feature matcher to describe which best match to keep for a given feature point. According to OpenCV's documentation, this parameter is used to check whether pairs of matches are dissimilar enough to be compared to each other:
  > Features matcher, which finds two best matches for each feature and leaves the best one only if the ratio between descriptor distances is greater than the threshold match_conf.

  In the `stitching` package, the default value for ORB features is `0.3`, for all other descriptors it is `0.65`

- `confidence_threshold` describes how confident we are that two images are from the same panorama. The parameter is used to filter out non-fitting images for the panorama result. Also, it is used to parameterize the bundle adjuster. In the `stitching` package, the default value is `1.0`.

**Reply by neriniccolodei** [reply]
Hello, thank you for the clarification. About `confidence_threshold`, I have seen this parameter called another way in `all_relevant_matches = matcher.draw_matches_matrix`, where the parameter is called `conf_thresh`. Meanwhile, the `Subsetter `class uses `confidence_threshold`. Is there a difference between these two? 

Should one tweak the 0.3 `match_conf` parameter or the `confidence_threshold` parameter?

**Reply by lukasalexanderweber** [reply]
this is just a naming issue. You could rename `conf_thresh` to `confidence_threshold`

**Reply by neriniccolodei** [reply]
Thank you. So, in the tutorial code if I change one parameter I need to change the other one accordingly, otherwise things are a bit clashing? Meaning changing `conf_thresh` within `matcher.draw_matches_matrix` and also within `Subsetter`.

**Reply by lukasalexanderweber** [reply]
Yes, you need to change both :)
